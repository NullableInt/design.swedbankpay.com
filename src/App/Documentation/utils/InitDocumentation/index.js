import React from "react";

import { ComponentPreview, Property } from "#";

const InitDocumentation = ({ componentName, scriptName }) => {
    // Ensures that the name is in lowercase [AW]
    const name = componentName ? componentName.toLowerCase() : "component";
    const script = scriptName ? scriptName.toLowerCase() : name;

    return (
        <>
            <h3>Initializing {name}s</h3>
            <p>
                To initialize {name}s use <Property value={`px.${script}.init()`} />. If <Property value=".init()" /> is called
                without arguments it will target all {name}s on the page. To initialize only a select few send one or more ids as an argument.
                The init function accepts both a single string or an array of strings. These strings need to match an
                existing {name} id and will print warning messages if an argument is invalid.
            </p>
            <hr/>
            <p>
                Example of valid arguments
            </p>
            <ComponentPreview language="javascript" codeFigure >
                {`px.${script}.init();\n`}
                {`px.${script}.init("demo-${script}");\n`}
                {`px.${script}.init(["demo-${script}"]);\n`}
                {`px.${script}.init(["demo-${script}-1", .. "demo-${script}-x"]);\n`}
            </ComponentPreview>
        </>
    );
};

export default InitDocumentation;
