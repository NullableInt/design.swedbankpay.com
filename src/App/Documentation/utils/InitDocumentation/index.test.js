import React from "react";
import { shallow } from "enzyme";

import InitDocumentation from "./index";

describe("Utilities: InitDocumentation", () => {
    it("is defined", () => {
        expect(InitDocumentation).toBeDefined();
    });

    it("renders with the correct component name", () => {
        const wrapper = shallow(<InitDocumentation componentName="Test" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains("test")).toEqual(true);
    });

    it("renders with a default component name set to Component", () => {
        const wrapper = shallow(<InitDocumentation />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains("component")).toEqual(true);
    });

    test.todo("write test for scriptName prop");
});

