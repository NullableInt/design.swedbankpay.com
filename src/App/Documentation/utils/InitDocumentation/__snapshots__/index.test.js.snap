// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities: InitDocumentation renders with a default component name set to Component 1`] = `
<Fragment>
  <h3>
    Initializing 
    component
    s
  </h3>
  <p>
    To initialize 
    component
    s use 
    <Property
      value="px.component.init()"
    />
    . If 
    <Property
      value=".init()"
    />
     is called without arguments it will target all 
    component
    s on the page. To initialize only a select few send one or more ids as an argument. The init function accepts both a single string or an array of strings. These strings need to match an existing 
    component
     id and will print warning messages if an argument is invalid.
  </p>
  <hr />
  <p>
    Example of valid arguments
  </p>
  <ComponentPreview
    codeFigure={true}
    language="javascript"
  >
    px.component.init();

    px.component.init("demo-component");

    px.component.init(["demo-component"]);

    px.component.init(["demo-component-1", .. "demo-component-x"]);

  </ComponentPreview>
</Fragment>
`;

exports[`Utilities: InitDocumentation renders with the correct component name 1`] = `
<Fragment>
  <h3>
    Initializing 
    test
    s
  </h3>
  <p>
    To initialize 
    test
    s use 
    <Property
      value="px.test.init()"
    />
    . If 
    <Property
      value=".init()"
    />
     is called without arguments it will target all 
    test
    s on the page. To initialize only a select few send one or more ids as an argument. The init function accepts both a single string or an array of strings. These strings need to match an existing 
    test
     id and will print warning messages if an argument is invalid.
  </p>
  <hr />
  <p>
    Example of valid arguments
  </p>
  <ComponentPreview
    codeFigure={true}
    language="javascript"
  >
    px.test.init();

    px.test.init("demo-test");

    px.test.init(["demo-test"]);

    px.test.init(["demo-test-1", .. "demo-test-x"]);

  </ComponentPreview>
</Fragment>
`;
